pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -170,9 +171,100 @@ func getPersistentVolumeClaimRetentionPolicy(set *apps.StatefulSet) apps.Statefu
 	return policy
 }
 
-// claimOwnerMatchesSetAndPod returns false if the ownerRefs of the claim are not set consistently with the
+// matchesRef returns true when the object matches the owner reference, that is the name and GVK are the same.
+func matchesRef(ref *metav1.OwnerReference, obj metav1.Object, gvk schema.GroupVersionKind) bool {
+	return gvk.GroupVersion().String() == ref.APIVersion && gvk.Kind == ref.Kind && ref.Name == obj.GetName()
```

Comment:
Why not using `IsControlledBy` from `k8s.io/apimachinery/pkg/apis/meta/v1`

Reply:
That matches on the UID, not on the GVK & name. So if, eg, the statefulset was updated, IsControlledBy won't return true until the UID is updated?